### PR TITLE
feat: table cell sparklines

### PR DESCRIPTION
**Pull request by pauldambra**
## Problem

I have a soft-spot for sparklines and saw we have one embedded in the ingestion warnings table.

## Changes

* makes that into a separate component
* allows data table nodes to specify column formats
* allows HogQL queries to prompt formats with their column alias (which is a bit yuck but not terrible)

## How did you test this code?

👀 locally

```
{
  "kind": "DataTableNode",
  "source": {
    "kind": "HogQLQuery",
    "query": "select event, (select [1,2,3]) as array_of_data, (select [1,2,3]) as bare_array from events where timestamp > now () - interval 1 day group by event limit 5"
  },
  "columnFormats": {
    "array_of_data": "sparkline"
  }
}
```

Screenshot 2023-06-15 at 19 45 48

```
{
  "kind": "DataTableNode",
  "source": {
    "kind": "HogQLQuery",
    "query": "select event, (select [1,2,3]) as array_of_data, (select [1,2,3]) as bare_array from events where timestamp > now () - interval 1 day group by event limit 5"
  },
  "columnFormats": {
    "array_of_data": "sparkline"
  }
}
```

Screenshot 2023-06-15 at 19 49 22

### CODE REVIEW

**Comment by mariusandra** [comment]
Looking cool, though this will be hard for any user to use, since they can't edit the (self-admittedly yuk) types array when making a query.

What if instead, we do:

```sql
select sparkline([1,2,3,4,5,6,7,8,9,10])
```

and it would just work? ✨🌈🦄 

It's easier than you think:

1. Add `sparkline` to `constants.py` so the function could be called
2. In `printer.py`, we employ a bunch of tricks and customisations [for a few functions](https://github.com/PostHog/posthog/blob/7e0d22b0fb296b3c2fd7f81331c7e24cde1d719b/posthog/hogql/printer.py#L464-L499). Add an exception for `sparkline` as well. 
3. It should just transform `sparkline([1,2,3])` into something like `map('__hogql_graph', 'sparkline', results, [1,2,3])` 
4. In the frontend, when printing a object that has `__hogql_graph` as its first argument, print a graph.

**Comment by pauldambra** [comment]
So now I can do 

```
SELECT
    error,
    sparkline(arrayMap(h -> countEqual(groupArray(hour), h), range(0,23))),
    count() as error_count
FROM
(
    SELECT
        properties.$exception_message as error,
        toHour(timestamp) AS hour
    FROM
        events
    WHERE
         timestamp > now () - interval 1 day
         and event = '$exception'
) subquery
GROUP BY
    error
ORDER BY
    error_count desc
```

to get

Screenshot 2023-06-15 at 21 51 53

File: frontend/src/lib/lemon-ui/LemonTable/TableCellSparkline.tsx [hunk 1]
```diff
@@ -1,49 +1,36 @@
...
 import { offset } from '@floating-ui/react'
 
-import './WarningEventsGraph.scss'
+import './TableCellSparkline.scss'
 
-export function WarningEventsGraph({ summary }: { summary: IngestionWarningSummary }): JSX.Element {
+export interface SparklineDataset {
+    dates?: string[]
+    data: number[]
+}
```
Comment (mariusandra) [inline frontend/src/lib/lemon-ui/LemonTable/TableCellSparkline.tsx:11]: I think these should be the component's props. I don't really see why we need to pass them through a `dataset` property. Plus, there's a re-rendering issue like this, partly my fault... see next comment.

File: frontend/src/queries/nodes/DataTable/renderColumn.tsx [hunk 2]
```diff
@@ -58,6 +59,16 @@ export function renderColumn(
         }
         if (typeof value === 'object') {
             if (Array.isArray(value)) {
+                if (value[0] === '__hogql_chart_type' && value[1] === 'sparkline') {
+                    const object: Record<string, any> = {}
+                    for (let i = 0; i < value.length; i += 2) {
+                        object[value[i]] = value[i + 1]
+                    }
+                    if ('results' in object && Array.isArray(object.results)) {
+                        return <TableCellSparkline dataset={{ data: object.results }} />
```
Comment (mariusandra) [inline frontend/src/queries/nodes/DataTable/renderColumn.tsx:68]: Every time we pass `{ data: object.results }` as `dataset`, we're creating a new object. Thus the `useEffect(..., [dataset])` code in sparkline runs every time the component is rendered. Probably better to pass `results` as a prop itself.

**Review by mariusandra — APPROVED** [review]
Just one remark reading simplifying props... but otherwise, let's get it in and have some fun 😍

**Pull request merged**